src/riak_kv_select.erl

```diff
@@ -117,19 +134,36 @@ downgrade_select([v2,v1=To|Tail], Select1) ->
         cover_context = Select1#riak_select_v2.cover_context,
         local_key     = Select1#riak_select_v2.local_key
     },
+    downgrade_select([To|Tail], Select2);
+downgrade_select([v3,v2=To|Tail], Select1) ->
+    Select2 = #riak_select_v2{
+        'SELECT'      = Select1#riak_select_v3.'SELECT',
```

Comment:
Sorry, this is not acceptable - the system cannot work like this where the position of a key in a record at compile time changes the behaviour...